src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_ReturnKey.cs

```diff
@@ -79,15 +84,17 @@ private void CommitOnEnter(out bool sendThrough, out bool committed)
                 return;
             }
 
-            // Get the text that the user has currently entered into the buffer
-            var viewSpan = model.GetSubjectBufferFilterSpanInViewBuffer(selectedItem.FilterSpan);
-            var textTypedSoFar = model.GetCurrentTextInSnapshot(
-                viewSpan, this.TextView.TextSnapshot, this.GetCaretPointInViewBuffer());
+            if (sendThrough)
+            {
+                // Get the text that the user has currently entered into the buffer
+                var viewSpan = model.GetSubjectBufferFilterSpanInViewBuffer(selectedItem.FilterSpan);
+                var textTypedSoFar = model.GetCurrentTextInSnapshot(
+                    viewSpan, this.TextView.TextSnapshot, this.GetCaretPointInViewBuffer());
+                sendThrough = selectedItem.CompletionProvider.SendEnterThroughToEditor(selectedItem, textTypedSoFar);
+            }
```

Comment:
Updated so we don't sendThrough if "Add new line on enter after end of fully typed word" is checked...